x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingContext.java

```diff
@@ -177,7 +177,7 @@ public boolean isEmpty() {
         return anomalyDetectionTasks.isEmpty()
             && snapshotUpgradeTasks.isEmpty()
             && dataframeAnalyticsTasks.isEmpty()
-            && modelAssignments.values().stream().allMatch(assignment -> assignment.totalTargetAllocations() == 0);
+            && modelAssignments.values().stream().allMatch(assignment -> assignment.getTaskParams().getNumberOfAllocations() == 0);
```

Comment:
Oh because if there are no ml nodes in the cluster `assignment.totalTargetAllocations() == 0` even if `assignment.getTaskParams().getNumberOfAllocations()  > 0`

Reply:
Yes, indeed